core/src/erasure.rs

```diff
@@ -380,18 +225,14 @@ pub mod test {
                 coding_blocks.iter_mut().map(|x| x.as_mut_slice()).collect();
             let v_slices: Vec<_> = vs.iter().map(|x| x.as_slice()).collect();
 
-            assert!(generate_coding_blocks(
-                coding_blocks_slices.as_mut_slice(),
-                v_slices.as_slice(),
-            )
-            .is_ok());
+            assert!(encode(v_slices.as_slice(), coding_blocks_slices.as_mut_slice(),).is_ok());
         }
         trace!("test_coding: coding blocks:");
         for b in &coding_blocks {
             trace!("test_coding: {:?}", b);
         }
-        let erasure: i32 = 1;
-        let erasures = vec![erasure, -1];
+        let erasure: usize = 1;
+        let erasures = vec![erasure];
         // clear an entry
         vs[erasure as usize].copy_from_slice(zero_vec.as_slice());
```

Comment:
this cast is unnecessary now?

Reply:
Yes, I just use `usize` erasures, convert to `i32` and append `-1` inside the jerasure encode that calls FFI functions. The rust erasure library takes `usize` anyway so no conversion will be needed there.